compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -460,9 +460,13 @@ impl<'f, 'tcx> Coerce<'f, 'tcx> {
         // to the target type), since that should be the least
         // confusing.
         let Some(InferOk { value: ty, mut obligations }) = found else {
-            let err = first_error.expect("coerce_borrowed_pointer had no error");
-            debug!("coerce_borrowed_pointer: failed with err = {:?}", err);
-            return Err(err);
+            if let Some(first_error) = first_error {
```

Comment:
This unwrap fails b/c above we simply do not see any types to autoderef, since structurally resolving the first autoderef of `&<not wf alias>` simply fails.